dts/riscv/hpmicro/hpm6750.dtsi

```diff
@@ -0,0 +1,246 @@
+			d-cache-line-size = <32>;
+			CPU0_intc: interrupt-controller {
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				compatible = "riscv,cpu-intc";
+			};
+		};
+		CPU1: cpu@1 {
+			compatible = "riscv";
+			device_type = "cpu";
+			reg = <1>;
+			status = "okay";
+			riscv,isa = "rv32imafdcxandes";
+			clock-frequency = <816000000>;
+			i-cache-line-size = <32>;
+			d-cache-line-size = <32>;
+			CPU1_intc: interrupt-controller {
+				#interrupt-cells = <1>;
+				interrupt-controller;
+				compatible = "riscv,cpu-intc";
+			};
+		};
+
+		FGPIO: peripheral@C0000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0xC0000 0x2000>;
+		};
+	};
+
+	soc {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		compatible = "hpmicro,hpm6750";
+		ranges;
+
+		mtimer: machine-timer@e6000000 {
+			compatible = "riscv,machine-timer";
+			reg = <0xe6000000 0x1
+			       0xe6000008 0x1>;
+			clk-divider = <RISCV_MACHINE_TIMER_DIVIDER_1>;
+		};
+
+		ilm: memory@0 {
+			compatible = "zephyr,memory-region";
+			reg = <0x0 DT_SIZE_K(256)>;
+			zephyr,memory-region = "ITCM";
+		};
+
+		dlm: memory@80000 {
+			compatible = "zephyr,memory-region";
+			reg = <0x080000 DT_SIZE_K(256)>;
+			zephyr,memory-region = "DTCM";
+		};
+
+		sram: memory@1080000 {
+			device_type = "memory";
+			compatible = "mmio-sram";
+			reg = <0x01080000 DT_SIZE_K(1024)>;
+		};
+
+		plic0: interrupt-controller@e4000000 {
+			compatible = "sifive,plic-1.0.0";
+			#address-cells = <1>;
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			reg = <	0xe4000000 0x00001000
+				0xe4002000 0x00000800
+				0xe4200000 0x00010000 >;
+			reg-names = "prio", "irq_en", "reg";
+			riscv,max-priority = <255>;
+			riscv,ndev = <1023>;
+			interrupts-extended = <&CPU0_intc 0 &CPU1_intc 0>;
+		};
+
+		plic_sw0: interrupt-controller@e6400000 {
+			compatible = "andestech,plic_sw";
+			#address-cells = <1>;
+			#interrupt-cells = <2>;
+			interrupt-controller;
+			reg = <0xe6400000 0x00400000>;
+			riscv,max-priority = <255>;
+			riscv,ndev = <1023>;
+			interrupts-extended = <&CPU0_intc 3 &CPU1_intc 3>;
+		};
+
+		gpio0: peripheral@f0000000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0xf0000000 0x4000>;
+		};
+
+		gpio1: peripheral@f0004000 {
+			#address-cells = <1>;
+			#size-cells = <1>;
+			ranges = <0x0 0xf0004000 0x4000>;
+		};
+
+		uart0: serial@f0040000 {
+			compatible = "hpmicro,hpm-uart";
+			label = "UART_0";
+			reg = <0xf0040000 0x1000>;
+			interrupts = <31 1>;
+			interrupt-parent = <&plic0>;
+			status = "disabled";
+		};
+
+		pinctrl: pin-controller@f4040000 {
+			compatible = "hpmicro,hpm-pinctrl";
+			reg = <0xf4040000 0x40000>;
+			status = "okay";
+			label = "PINCTRL";
+		};
+	};
+};
+
+&gpio0 {
+	gpioa: gpio@0 {
+		compatible = "hpmicro,hpm-gpio";
+		reg = <0x0 0x4000>;
+		gpio-controller;
+		#gpio-cells = <2>;
+		hpmicro-gpio-port = <0>;
+		interrupts = <1 1>;
+		interrupt-parent = <&plic0>;
+		label = "GPIOA";
+	};
```

Comment:
why not define these entries after `gpio0` definition above?

Reply:
On this chip gpioa-gpioz can be assigned to any of the gpio controllers gpio0, gpio1 and fpgio.